src/lib/statistics/table_statistics.cpp

```diff
@@ -238,6 +238,25 @@ TableStatistics TableStatistics::estimate_predicated_join(const TableStatistics&
   };
 
   switch (mode) {
+    case JoinMode::Semi:
+      join_table_stats._column_statistics[column_ids.first] = stats_container.left_column_statistics;
+
+      // Simple heuristic: we assume that three quarters of the element in the smaller relation
+      // (we are upper-bound by number of non-null values in both relations) will match.
+      join_table_stats._row_count =
+          0.75 * std::min(row_count() * stats_container.left_column_statistics->non_null_value_ratio(),
+                          right_table_statistics.row_count() *
+                              stats_container.right_column_statistics->non_null_value_ratio());
+      break;
```

Comment:
I don't think the current table statistics will help us to come up with a usable solution. Voting to post-pone accurate estimation until is comes up as a problem (currently, it will be neglected in the join ordering anyways since it is not an inner join).